airflow/providers/salesforce/hooks/salesforce.py

```diff
@@ -236,6 +236,61 @@ def write_object_to_file(self,
         if fmt not in ['csv', 'json', 'ndjson']:
             raise ValueError("Format value is not recognized: {}".format(fmt))
 
+        if filename is None:
+            raise ValueError("Filename cannot be None.")
```

Comment:
I don't see why this should be validated. The type of variable is `str`. So this can not be `None` if you use it correctly.

Reply:
A type of `Optional[str]` can be `None`.

Reply:
Added it as an additional check, removed it as the argument is not optional.